test/integration/test_phone_number_lookup.py

```diff
@@ -0,0 +1,168 @@
+        
+        get_lookup_status_response: LookupStatus = self.api_instance.get_lookup_status(self.account_id, create_lookup_response[0].request_id, _return_http_data_only=False)
+        self.assertEqual(get_lookup_status_response[1], 200)
+
+        # make the request again without _return_http_data_only=False to just get the LookupStatus model
+        get_lookup_status_response: LookupStatus = self.api_instance.get_lookup_status(self.account_id, create_lookup_response[0].request_id)
+        get_lookup_status_response_attempts = 1
+        while get_lookup_status_response.status != LookupStatusEnum('COMPLETE'): 
+            # Raise an error if it takes more than 5 requests to get COMPLETE status
+            if get_lookup_status_response_attempts == 5:
+                raise Exception(f'Took too long to get phone number lookup \'COMPLETE\' status. Aborting test after {get_lookup_status_response_attempts} attempts.')
+            time.sleep(2)
+
+            get_lookup_status_response: LookupStatus = self.api_instance.get_lookup_status(self.account_id, create_lookup_response[0].request_id)
+            get_lookup_status_response_attempts += 1
+
+        self.assertEqual(get_lookup_status_response.request_id, create_lookup_response[0].request_id)
+        self.assertIs(type(get_lookup_status_response), LookupStatus)
+        
+        # Assert that each result is of type LookupResult
+        for i in range(len(get_lookup_status_response.result)):
+            self.assertIs(type(get_lookup_status_response.result[i]), LookupResult)
+
+        # Check the information for a Bandwidth TN
+        bw_lookup_result: LookupResult = get_lookup_status_response.result[0]
+        self.assertEqual(bw_lookup_result.response_code, 0)
+        self.assertIs(type(bw_lookup_result.message), str)
+        self.assertEqual(bw_lookup_result.e_164_format, os.environ['BW_NUMBER'])
+        self.assertIs(type(bw_lookup_result.formatted), str)
+        self.assertTrue(bw_lookup_result.country == "US" or bw_lookup_result.country == "Canada")
+        self.assertTrue(bw_lookup_result.line_type == "Mobile" or bw_lookup_result.line_type == "Fixed")
+        self.assertIn("Bandwidth", bw_lookup_result.line_provider)
+
+        # Check the information for a Verizon(true mobile) TN
+        vzw_lookup_result = get_lookup_status_response.result[1]
+        self.assertEqual(vzw_lookup_result.response_code, 0)
+        self.assertIs(type(vzw_lookup_result.message), str)
+        self.assertEqual(vzw_lookup_result.e_164_format, os.environ['VZW_NUMBER'])
+        self.assertIs(type(vzw_lookup_result.formatted), str)
+        self.assertTrue(vzw_lookup_result.country == "US" or vzw_lookup_result.country == "Canada")
+        self.assertTrue(vzw_lookup_result.line_type == "Mobile" or vzw_lookup_result.line_type == "Fixed")
+        self.assertIn("Verizon", vzw_lookup_result.line_provider)
+        self.assertIs(type(vzw_lookup_result.mobile_country_code), str)
```

Comment:
The assertions provided by `unitest` library seem to be very basic ones. We can benefit from using other libraries like [assertpy](https://github.com/assertpy/assertpy) and [pyhamcrest](https://github.com/hamcrest/PyHamcrest). Both are inspired from Java. This might make writing Tests a bit more enjoyable! Take a look and see what you think.

Reply:
Will take a look - id be in favor of those or using pytest 

Will have to look into how to get those into the requirements file and ensure it isnt overwritten when we re-generate (but should be doable easily)